File: Backend/TripPlannerBackend.API/Entity/Activity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripPlannerBackend.API.Entity
{
    public class Activity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public string? Description { get; set; }
        public string Location { get; set; }
        public int TripId { get; set; }
        public Trip Trip { get; set; }
        public ActivityType ActivityType { get; set; }
        [ForeignKey("Cost")]
        public int? CostId { get; set; }
        public Cost? Cost { get; set; }
    }
}

File: Frontend/src/app/model/Activity.ts
import { ActivityType } from "./ActivityType";
import { Cost } from "./Cost";

export interface Activity {
    id: number;
    name: string;
    activityType: ActivityType;
    cost? : Cost;
    start: Date;
    end: Date;
    description?: string;
    location: string;
    // location: Location;
}